vmdb/app/models/miq_report/search.rb

```diff
@@ -14,7 +14,7 @@ def get_limit_offset(page, per_page)
   end
 
   def get_sqltable(assoc)
-    r = self.db_class.reflect_on_association(assoc.to_sym)
+    r = db_class.reflection_with_virtual(assoc.to_sym)
```

Comment:
Yes, that's correct.  `reflect_on_association` in `ActsAsArModel` [would consult the virtual reflections](https://github.com/tenderlove/manageiq/blob/2c4dc077c072b3563e8ce0a309dc64f4ebc3efcd/vmdb/lib/acts_as_ar_model.rb#L117-119).  On real models, we need to call `reflection_with_virtual` to include virtual reflections in the search.